x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizerTests.java

```diff
@@ -366,6 +412,27 @@ public void testCombineProjectionWithAggregation() {
         assertThat(Expressions.names(agg.groupings()), contains("last_name", "first_name"));
     }
 
+    /**
+     * Expects
+     * Limit[1000[INTEGER]]
+     * \_Aggregate[[last_name{f}#23, first_name{f}#20, k{r}#4],[SUM(salary{f}#24) AS s, last_name{f}#23, first_name{f}#20, first_n
+     * ame{f}#20 AS k]]
```

Comment:
Potentially out of scope, but: couldn't/shouldn't we prune `k` from the groupings, since it's an alias for `first_name`? `k` doesn't seem to be used in the aggregates, either.

Reply:
See https://github.com/elastic/elasticsearch/issues/107166